### PR TITLE
Add support for closing threads

### PR DESCRIPTION
**Pull request by redcreeper14385**

### CODE REVIEW

**Review by DenverCoder1 — COMMENTED** [review]
Should manage_messages be required? Maybe the creator of the thread should be able to close it?

File: cogs/help.py [hunk 1]
```diff
@@ -40,11 +40,17 @@ async def on_message(self, message):
...
     @commands.command()
     @commands.is_owner()
     async def help_menu(self, ctx):
         await ctx.send("Click a button to create a help thread!", view=HelpView())
 
+    @commands.command()
+    @commands.has_permissions(manage_messages=True)
+    async def close(self, ctx):
+        if isinstance(ctx.channel, nextcord.Thread) and ctx.channel.parent_id == self.help_channel:
+            ctx.channel.archived = True
```
Comment (TAG-Epic) [inline cogs/help.py:52]: I'm pretty sure this won't work.

File: cogs/help.py [hunk 2]
```diff
@@ -40,11 +40,17 @@ async def on_message(self, message):
...
             await message.delete(delay=5)
 
-
     @commands.command()
     @commands.is_owner()
     async def help_menu(self, ctx):
         await ctx.send("Click a button to create a help thread!", view=HelpView())
 
+    @commands.command()
+    @commands.has_permissions(manage_messages=True)
```
Comment (TAG-Epic) [inline cogs/help.py:49]: This should be limited to thread creator and helpers.

**Review by TAG-Epic — CHANGES_REQUESTED** [review]
This should also log closes to a channel imo

File: cogs/help.py (hunk 1 above)
Comment (redcreeper14385) [reply cogs/help.py:52]: I couldn't find the correct method

File: cogs/help.py (hunk 2 above)
Comment (redcreeper14385) [reply cogs/help.py:49]: The thread creator is always the bot, I can make it helpers only though.

Comment (DenverCoder1) [reply cogs/help.py:49]: You should be able to determine who requested the thread to be opened. Worst case you can check the `mentions` in the first message.

Comment (redcreeper14385) [reply cogs/help.py]: Fixed in 5b52730.

File: cogs/help.py (hunk 1 above)
Comment (redcreeper14385) [reply cogs/help.py]: Fixed in 5b52730.

**Comment by redcreeper14385** [comment]
> This should also log closes to a channel imo

Fixed in 5711cb1

**Comment by redcreeper14385** [comment]
This PR will require the creation of a channel, `help_logs`.

File: cogs/help.py [hunk 3]
```diff
@@ -49,11 +49,14 @@ async def help_menu(self, ctx):
     async def close(self, ctx):
         history = ctx.channel.history(oldest_first=True, limit=1)
         history_flat = await history.flatten()
+        moderator_channel = nextcord.utils.get(ctx.guild.channels, name='help_logs')
```
Comment (DenverCoder1) [inline cogs/help.py]: Better to get by id, the name of the channel can change

**Pull request merged**